### ISSUE TITLE
import (absolute path) suggestions break in 1.17 with @typings installed

### ISSUE THREAD

**Issue by wesleymostien** [Microsoft/vscode#37132]
Problem: 

Path suggestions for absolute paths (configured in tsconfig) break for import statements in version 1.17 as soon as @typings are installed. 

Without the @typings folder anywhere in the project, everything works fine.
In version 1.16.1 it even works fine WITH the typings installed : 

![screen shot 2017-10-30 at 11 50 43](https://user-images.githubusercontent.com/6223285/32167597-a642dda6-bd69-11e7-98af-18efca098fd2.png)
 
But when I do for instance : 
`````
npm i @types/node
`````
after that is installed, this is what I get for suggestions instead of above screenshot (it seems that it only suggests installed types) : 

![screen shot 2017-10-30 at 11 50 18](https://user-images.githubusercontent.com/6223285/32167624-bc744ad8-bd69-11e7-9801-43229a40769a.png)

- VSCode Version: Code 1.17.2 (b813d12980308015bcd2b3a2f6efa5c810c33ba5, 2017-10-16T13:47:41.889Z)
- OS Version: Darwin x64 16.7.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
bracket-pair-colorizer|Coe|0.10.10
beautify|Hoo|1.1.1
material-icon-theme|PKi|2.2.4
path-intellisense|chr|1.4.2
vue-peek|dar|1.0.2
vscode-eslint|dba|1.4.0
ts-tslint|eg2|0.0.4
tslint|eg2|1.0.13
prettier-vscode|esb|0.24.0
docthis|joe|0.6.0
vetur|oct|0.10.1
typescript-hero|rbb|1.6.0
guides|spy|0.9.0
vscode-todo-highlight|way|0.5.11
JavaScriptSnippets|xab|1.5.0

(1 theme extensions excluded)

---

Steps to Reproduce:

1. configure absolute path for typescript project in tsconfig. eg : 
`````
{
  "compilerOptions": {
    "target": "es6",
    "lib": ["dom", "es2015", "es2017"],
    "module": "commonjs",
    "moduleResolution": "node",
    "baseUrl": "./",
    "paths": {
      "@*": ["./*"]
    }
  }
}
`````
2. start typing 
`````
import whatever from '@/'
`````

Reproduces without extensions: No (doesn't work either way)

**Comment by mjbvz** [comment]
I believe the original suggestions may have come from your path-intellisense extension. I say that because I don't know think we ever properly supported the `paths` config option for our built-in path intellisense (https://github.com/Microsoft/TypeScript/issues/19113)

Using [VS Code insiders](https://code.visualstudio.com/insiders/), can you try setting `"typescript.quickSuggestionsForPaths": false`. This should prevent our built-in path intellisense from automatically kicking in

**Comment by airato** [comment]
I've tried with latest VS Code Insiders – import path autocomplete does not recognize tsconfig's `paths` option correctly. Keys from the `paths` map appear as is, they are not being processed from globs to matches.
Tried with and without quickSuggestionsForPaths, path-intellisense extension.

Version 1.18.0-insider (1.18.0-insider), 2ab10d50163f008ada9a4aa215de8c884857ce99, 2017-11-01T06:07:36.569Z

```json
{
  "compilerOptions": {
    "target": "es6",
    "lib": ["dom", "es2015", "es2017"],
    "module": "commonjs",
    "moduleResolution": "node",
    "baseUrl": "./",
    "paths": {
      "~/*": ["src/*"]
    }
  }
}
```

![image](https://user-images.githubusercontent.com/4506749/32302488-37ba8272-bf20-11e7-8623-ba04243bc28e.png)

cc @MiguelMadero
